test/common/common/utility_test.cc

```diff
@@ -94,6 +94,78 @@ TEST(StringUtil, rtrim) {
   }
 }
 
+TEST(StringUtil, ltrim) {
+  {
+    std::string test("   ");
+    StringUtil::ltrim(test);
+    EXPECT_EQ("", test);
+  }
+
+  {
+    std::string test("   hello \r\t\r\n");
+    StringUtil::ltrim(test);
+    EXPECT_EQ("hello \r\t\r\n", test);
```

Comment:
fwiw if you write the methods functionally the tests will be more compact, e.g.

```
  EXPECT_EQ("hello  ", StringUtil::ltrim(" hello  "));
```

Reply:
👀